apps/namadillo/src/slices/validators.ts

```diff
@@ -34,50 +40,116 @@ export type MyValidator = {
+      const [validatorsResponse, votingPowerResponse] = await Promise.all([
+        api.apiV1PosValidatorGet(),
+        api.apiV1PosVotingPowerGet(),
+      ]);
+
+      // TODO: rename one data to items?
+      const validators = validatorsResponse.data.data;
+      const votingPower = votingPowerResponse.data;
 
-export const allValidatorsAtom = atomWithQuery((get) => ({
-  queryKey: ["all-validators"],
-  queryFn: async () => {
-    const rpcUrl = get(rpcUrlAtom);
-    const query = new Query(rpcUrl);
-    const queryResult =
-      (await query.query_all_validator_addresses()) as string[];
-    return queryResult.map(toValidator);
-  },
-}));
+      return validators.map((v) => toValidator(v, votingPower, parameters));
+    },
+  };
+});
 
 // eslint-disable-next-line
 export const myValidatorsAtom = atomWithQuery((get) => {
   const account = get(defaultAccountAtom);
+  const chainParameters = get(chainParametersAtom);
 
   // TODO: Refactor after this event subscription is enabled in the indexer
   const enablePolling = get(shouldUpdateBalanceAtom);
 
   return {
-    queryKey: ["my-validators", account.data?.address],
+    queryKey: [
+      "my-validators",
+      account.data?.address,
+      chainParameters.dataUpdatedAt,
+    ],
     enabled: account.isSuccess,
     refetchInterval: enablePolling ? 1000 : false,
     queryFn: async (): Promise<MyValidator[]> => {
-      const rpcUrl = get(rpcUrlAtom);
-      const query = new Query(rpcUrl);
-      const myValidatorsRes = await query.query_my_validators([
-        account.data?.address,
+      const unbondingPeriod =
+        chainParameters.data?.unbondingPeriodInDays || BigInt(0);
+      const api = new DefaultApi();
+      const [bondsResponse, totalVotingPowerResponse] = await Promise.all([
+        // TODO: is address always available?
+        api.apiV1PosBondAddressGet(account.data!.address!),
+        api.apiV1PosVotingPowerGet(),
       ]);
-      return myValidatorsRes.map(toMyValidators);
+
+      return toMyValidators(
+        bondsResponse.data.data,
+        totalVotingPowerResponse.data,
+        unbondingPeriod
+      );
+    },
+  };
+});
+
+export const myUnbondsAtom = atomWithQuery<MyValidator[]>((get) => {
+  const chainParameters = get(chainParametersAtom);
+  const account = get(defaultAccountAtom);
+
+  // TODO: Refactor after this event subscription is enabled in the indexer
+  const enablePolling = get(shouldUpdateBalanceAtom);
+  return {
+    refetchInterval: enablePolling ? 1000 : false,
+    queryKey: [
+      "my-unbonds",
+      account.data?.address,
+      chainParameters.dataUpdatedAt,
+    ],
+    queryFn: async () => {
+      const unbondingPeriod =
+        chainParameters.data?.unbondingPeriodInDays || BigInt(0);
+      const api = new DefaultApi();
+      const [unbondsResponse, totalVotingPowerResponse] = await Promise.all([
+        // TODO: is address always available?
+        api.apiV1PosUnbondAddressGet(account.data!.address!),
```

Comment:
Same here, if separated in two different atoms, we would save this repeated query here